src/ray/rpc/grpc_server.cc

```diff
@@ -59,19 +59,27 @@ void GrpcServer::Run() {
     }
     // Build and start server.
     server_ = builder.BuildAndStart();
-    if (port_ > 0) {
+    if (server_) {
       break;
     }
     std::this_thread::sleep_for(std::chrono::milliseconds(
         RayConfig::instance().grpc_server_retry_timeout_milliseconds()));
     num_retries--;
   }
 
-  // If the grpc server failed to bind the port, the `port_` will be set to 0.
-  RAY_CHECK(port_ > 0)
-      << "Port " << specified_port
-      << " specified by caller already in use. Try passing node_manager_port=... into "
-         "ray.init() to pick a specific port";
+  std::ostringstream ss;
+  if (!server_) {
+    ss << "Failed to start the grpc server.";
+    if (specified_port != 0) {
+      ss << " Specified port is " << specified_port
+         << "and maybe it's already in use. Try passing node_manager_port=... into "
+            "ray.init() to pick a specific port";
+    } else {
+      ss << " No specified port.";
```

Comment:
This error seems to be pretty ambiguous. Can you make it a bit clearer what's the problem here?

Reply:
Maybe I'll just remove the port part message because grpc already outputs the reason?